src/requests/adapters.py

```diff
@@ -481,20 +487,25 @@ def send(
         else:
             timeout = TimeoutSauce(connect=timeout, read=timeout)
 
+        urlopen_kwargs = self.urllib3_response_options.copy()
+        urlopen_kwargs.update(
+            {
+                "method": request.method,
+                "url": url,
+                "body": request.body,
+                "headers": request.headers,
+                "redirect": False,
+                "assert_same_host": False,
+                "preload_content": False,
+                "decode_content": False,
+                "retries": self.max_retries,
+                "timeout": timeout,
+                "chunked": chunked,
+            }
+        )
+
```

Comment:
I'm wondering if this should be done in the opposite order (defaults updated by `self.urllib3_response_options`). If we add this feature, we'll almost certainly get future issues with questions like "Why can't I change `preload_content` or `decode_content` with the \<variable name\>?".

If we're going to expose something like this to avoid custom `send` implementations, it should probably be the full set of arguments. We can still keep the same defaults, but whatever users want to pass us, we should just forward.

Reply:
In my experience if there's something like this designed to help people access something as an escape hatch it will be misunderstood and misused. Copying the dictionary and updating it will avoid footguns. No one will override the headers, method, data, or anything else. I'm not particularly strongly invested in this feature as a whole but the other way I think we can enable this (because overriding send is a nightmare) is breaking out the actual call to urlopen into a different method that can be overridden.